pertpy/tools/_differential_gene_expression/_base.py

```diff
@@ -449,20 +506,437 @@ def _map_genes_categories_highlight(
+    ) -> Figure | None:
+        """Creates a pairwise expression plot from a pandas DataFrame or Anndata.
+
+        Makes on panel with a paired scatterplot for each variable.
+
+        Args:
+            adata: AnnData object, can be pseudobulked.
+            var_names: Variables to plot.
+            groupby: Column in adata.obs containing the grouping. Must contain exactly two different values.
+            pairedby: Column in adata.obs containing the pairing (e.g. "patient_id"). If None, an independent t-test is performed.
+            hue: Column in adata.obs to color by.
+            layer: Layer to use for plotting.
+            n_cols: Number of columns in the plot.
+            panel_size: Size of each panel.
+            show_legend: Whether to show the legend.
+            size: Size of the points.
+            y_label: Label for the y-axis.
+            pvalues: P-values for each variable. If None, they are calculated.
+            pvalue_template: Template for the p-value string displayed in the title of each panel.
+            adjust_fdr: Whether to correct p-values for false discovery rate.
+            boxplot_properties: Additional properties for the boxplot, passed to seaborn.boxplot.
+            palette: Color palette for the line- and stripplot.
+            {common_plot_args}
+
+        Returns:
+            If `return_fig` is `True`, returns the figure, otherwise `None`.
+
+        Examples:
+            >>> # Example with EdgeR
+            >>> import pertpy as pt
+            >>> adata = pt.dt.zhang_2021()
+            >>> adata.layers["counts"] = adata.X.copy()
+            >>> ps = pt.tl.PseudobulkSpace()
+            >>> pdata = ps.compute(
+            ...     adata,
+            ...     target_col="Patient",
+            ...     groups_col="Cluster",
+            ...     layer_key="counts",
+            ...     mode="sum",
+            ...     min_cells=10,
+            ...     min_counts=1000,
+            ... )
+            >>> edgr = pt.tl.EdgeR(pdata, design="~Efficacy+Treatment")
+            >>> edgr.fit()
+            >>> res_df = edgr.test_contrasts(
+            ...     edgr.contrast(column="Treatment", baseline="Chemo", group_to_compare="Anti-PD-L1+Chemo")
+            ... )
+            >>> edgr.plot_paired(pdata, var_names=res_df["variable"][:8], groupby="Treatment", pairedby="Major celltype")
+
+        Preview:
+            .. image:: /_static/docstring_previews/de_paired_expression.png
+        """
+        if boxplot_properties is None:
+            boxplot_properties = {}
+        groups = adata.obs[groupby].unique()
+        if len(groups) != 2:
+            raise ValueError("The number of groups in the group_by column must be exactly 2 to enable paired testing")
+        if pairedby is not None and hue is not None and (adata.obs.groupby(pairedby)[hue].nunique().max() > 1):
+            raise ValueError("Each paired sample must have an unambiguous hue")
+
+        adata = adata[:, var_names]
```

Comment:
This triggers a warning, right?

Reply:
I’ve just tested it, but it didn’t raise a warning for me. Do you have any resources on this?

Reply:
Nevermind then. Thought it's about a view issue.